File: quickSort.py
def quickSort(arr):
  if len(arr) < 2:
    return arr
  else: 
    pivot = arr[0]
    less = [i for i in arr[1:] if i <= pivot]
    greater = [i for i in arr[1:] if i >= pivot]
    return quickSort(less) + [pivot] + quickSort(greater)

print(quickSort([3,6,2,8,9,1,4,5]))
print(quickSort([]))
print(quickSort([1]))
print(quickSort([2,1]))

File: quickSort.js
function quickSort(arr) {
  if (arr.length <= 1) {
    console.log(arr);
    return arr
  } else {
    let pivot = arr[0],
      answer = [],
      less = [],
      greater = [];

    for (let i = 0; i < arr.length; i++) {
      if (arr[i] <= pivot) {
        less.push(arr[i])
      } else {
        greater.push(arr[i])
      }
    }
    answer.concat(quickSort(less), pivot, quickSort(greater));
    console.log(answer);
    return answer;
  }
}

quickSort([
  1,
  4,
  6,
  3,
  9,
  6,
  5
])

quickSort([1]);